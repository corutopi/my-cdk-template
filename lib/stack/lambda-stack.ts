import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';

import { IamRole } from '../resource/lambda/iam-role';
import { LambdaFunction } from '../resource/lambda/lambda-function';

/**
 * LambdakStack を作成するクラス.
 */
export class LambdakStack extends cdk.Stack {
  public readonly iamRole: IamRole;
  public readonly lambdaFunction: LambdaFunction;

  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    this.iamRole = new IamRole({ scope: this });
    this.lambdaFunction = new LambdaFunction({ scope: this }, { iamRole: this.iamRole });
  }
}
