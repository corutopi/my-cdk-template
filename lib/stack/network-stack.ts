import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';

import { BaseStack, BaseStackProps } from './abstruct/base-stack';
import { Vpc } from '../resource/network/vpc';
import { Subnet } from '../resource/network/subnet';
import { InternetGateway } from '../resource/network/internet-gateway';
import { RouteTable } from '../resource/network/route-table';

/**
 * NetworkStack を作成するクラス.
 */
export class NetworkStack extends BaseStack {
  public readonly vpc: Vpc;
  public readonly subnet: Subnet;
  public readonly internetGateway: InternetGateway;
  public readonly routeTable: RouteTable;

  constructor(parentProps: BaseStackProps) {
    super(parentProps);

    this.vpc = new Vpc({ scope: this });
    this.subnet = new Subnet({ scope: this }, { vpc: this.vpc });
    this.internetGateway = new InternetGateway({ scope: this }, { vpc: this.vpc });
    this.routeTable = new RouteTable(
      { scope: this },
      { vpc: this.vpc, subent: this.subnet, internetGateway: this.internetGateway }
    );
  }
}
