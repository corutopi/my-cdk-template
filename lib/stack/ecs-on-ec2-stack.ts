import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';

import { IamRole } from '../resource/ecs-on-ec2/iam-role';
import { SecurityGroup } from '../resource/ecs-on-ec2/security-group';
import { Instance } from '../resource/ecs-on-ec2/ec2-instance';
import { EcsCluster } from '../resource/ecs-on-ec2/ecs-cluster';
import { NetworkStack } from '../stack/network-stack';

/**
 * NetworkStack を作成するクラス.
 */
export class EcsOnEc2Stack extends cdk.Stack {
  public readonly role: IamRole;
  public readonly sg: SecurityGroup;
  public readonly ins: Instance;
  public readonly cluster: EcsCluster;

  constructor(scope: Construct, id: string, networkStack: NetworkStack, props?: cdk.StackProps) {
    super(scope, id, props);
    this.role = new IamRole({ scope: this });
    this.sg = new SecurityGroup({ scope: this }, { vpc: networkStack.vpc });
    this.ins = new Instance(
      { scope: this },
      { role: this.role, sg: this.sg, subnet: networkStack.subnet }
    );
    this.cluster = new EcsCluster({ scope: this });
  }
}
