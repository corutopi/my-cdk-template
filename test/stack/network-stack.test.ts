import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { NetworkStack } from '../../lib/stack/network-stack';

/**
 * テストで使用するテンプレートを生成する.
 *
 * @returns 生成した Template インスタンス
 */
function createTestTemplate(): Template {
  const app = new cdk.App();
  const stack = new NetworkStack(app, 'TestStack');
  const template = Template.fromStack(stack);
  return template;
}

test('Snapshot', () => {
  const template = createTestTemplate();
  const json = template.toJSON();

  expect(json).toMatchSnapshot('NetworkStackSnapshot');
});

test('Vpc', () => {
  const template = createTestTemplate();

  template.resourceCountIs('AWS::EC2::VPC', 1);
  template.hasResourceProperties('AWS::EC2::VPC', {
    CidrBlock: '10.10.0.0/16',
    Tags: [{ Key: 'Name', Value: 'cdktest-main-vpc' }],
  });
});

test('Subnet', () => {
  const template = createTestTemplate();

  template.resourceCountIs('AWS::EC2::Subnet', 4);
  template.hasResourceProperties('AWS::EC2::Subnet', {
    CidrBlock: '10.10.1.0/24',
    VpcId: { Ref: 'VpcMain' },
    Tags: [{ Key: 'Name', Value: 'cdktest-public-a-subnet' }],
  });
  template.hasResourceProperties('AWS::EC2::Subnet', {
    CidrBlock: '10.10.2.0/24',
    VpcId: { Ref: 'VpcMain' },
    Tags: [{ Key: 'Name', Value: 'cdktest-public-c-subnet' }],
  });
  template.hasResourceProperties('AWS::EC2::Subnet', {
    CidrBlock: '10.10.11.0/24',
    VpcId: { Ref: 'VpcMain' },
    Tags: [{ Key: 'Name', Value: 'cdktest-private-a-subnet' }],
  });
  template.hasResourceProperties('AWS::EC2::Subnet', {
    CidrBlock: '10.10.12.0/24',
    VpcId: { Ref: 'VpcMain' },
    Tags: [{ Key: 'Name', Value: 'cdktest-private-c-subnet' }],
  });
});

test('InternetGateway', () => {
  const template = createTestTemplate();

  template.resourceCountIs('AWS::EC2::InternetGateway', 1);
  template.hasResourceProperties('AWS::EC2::InternetGateway', {
    Tags: [{ Key: 'Name', Value: 'cdktest-main-igw' }],
  });
  template.resourceCountIs('AWS::EC2::VPCGatewayAttachment', 1);
  template.hasResourceProperties('AWS::EC2::VPCGatewayAttachment', {
    VpcId: { Ref: 'VpcMain' },
    InternetGatewayId: { Ref: 'InternetGatewayMain' },
  });
});

test('RouteTable', () => {
  const template = createTestTemplate();

  template.resourceCountIs('AWS::EC2::RouteTable', 2);
  template.hasResourceProperties('AWS::EC2::RouteTable', {
    VpcId: { Ref: 'VpcMain' },
    Tags: [{ Key: 'Name', Value: 'cdktest-public-common-rt' }],
  });
  template.hasResourceProperties('AWS::EC2::RouteTable', {
    VpcId: { Ref: 'VpcMain' },
    Tags: [{ Key: 'Name', Value: 'cdktest-private-common-rt' }],
  });

  template.resourceCountIs('AWS::EC2::Route', 1);
  template.hasResourceProperties('AWS::EC2::Route', {
    RouteTableId: { Ref: 'RouteTablePublicCommon' },
    DestinationCidrBlock: '0.0.0.0/0',
    GatewayId: { Ref: 'InternetGatewayMain' },
  });

  template.resourceCountIs('AWS::EC2::SubnetRouteTableAssociation', 4);
  template.hasResourceProperties('AWS::EC2::SubnetRouteTableAssociation', {
    RouteTableId: { Ref: 'RouteTablePublicCommon' },
    SubnetId: { Ref: 'SubnetPublicA' },
  });
  template.hasResourceProperties('AWS::EC2::SubnetRouteTableAssociation', {
    RouteTableId: { Ref: 'RouteTablePublicCommon' },
    SubnetId: { Ref: 'SubnetPublicC' },
  });
  template.hasResourceProperties('AWS::EC2::SubnetRouteTableAssociation', {
    RouteTableId: { Ref: 'RouteTablePrivateCommon' },
    SubnetId: { Ref: 'SubnetPrivateA' },
  });
  template.hasResourceProperties('AWS::EC2::SubnetRouteTableAssociation', {
    RouteTableId: { Ref: 'RouteTablePrivateCommon' },
    SubnetId: { Ref: 'SubnetPrivateC' },
  });
});
