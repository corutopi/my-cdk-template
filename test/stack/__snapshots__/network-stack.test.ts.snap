// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: NetworkStackSnapshot 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "InternetGatewayMain": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "cdktest-dev-main-igw",
          },
        ],
      },
      "Type": "AWS::EC2::InternetGateway",
    },
    "InternetGatewayMainAttachment": Object {
      "Properties": Object {
        "InternetGatewayId": Object {
          "Ref": "InternetGatewayMain",
        },
        "VpcId": Object {
          "Ref": "VpcMain",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
    "RouteTablePrivateCommon": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "cdktest-dev-private-common-rt",
          },
        ],
        "VpcId": Object {
          "Ref": "VpcMain",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "RouteTablePublicCommon": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "cdktest-dev-public-common-rt",
          },
        ],
        "VpcId": Object {
          "Ref": "VpcMain",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "RouteTablePublicRouteCommon": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "InternetGatewayMain",
        },
        "RouteTableId": Object {
          "Ref": "RouteTablePublicCommon",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "RouteTableSubnetAssociationPrivateA": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "RouteTablePrivateCommon",
        },
        "SubnetId": Object {
          "Ref": "SubnetPrivateA",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "RouteTableSubnetAssociationPrivateC": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "RouteTablePrivateCommon",
        },
        "SubnetId": Object {
          "Ref": "SubnetPrivateC",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "RouteTableSubnetAssociationPublicA": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "RouteTablePublicCommon",
        },
        "SubnetId": Object {
          "Ref": "SubnetPublicA",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "RouteTableSubnetAssociationPublicC": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "RouteTablePublicCommon",
        },
        "SubnetId": Object {
          "Ref": "SubnetPublicC",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "SubnetPrivateA": Object {
      "Properties": Object {
        "AvailabilityZone": "ap-northeast-1a",
        "CidrBlock": "10.10.11.0/24",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "cdktest-dev-private-a-subnet",
          },
        ],
        "VpcId": Object {
          "Ref": "VpcMain",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "SubnetPrivateC": Object {
      "Properties": Object {
        "AvailabilityZone": "ap-northeast-1c",
        "CidrBlock": "10.10.12.0/24",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "cdktest-dev-private-c-subnet",
          },
        ],
        "VpcId": Object {
          "Ref": "VpcMain",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "SubnetPublicA": Object {
      "Properties": Object {
        "AvailabilityZone": "ap-northeast-1a",
        "CidrBlock": "10.10.1.0/24",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "cdktest-dev-public-a-subnet",
          },
        ],
        "VpcId": Object {
          "Ref": "VpcMain",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "SubnetPublicC": Object {
      "Properties": Object {
        "AvailabilityZone": "ap-northeast-1c",
        "CidrBlock": "10.10.2.0/24",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "cdktest-dev-public-c-subnet",
          },
        ],
        "VpcId": Object {
          "Ref": "VpcMain",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VpcMain": Object {
      "Properties": Object {
        "CidrBlock": "10.10.0.0/16",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "cdktest-dev-main-vpc",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
